sql/src/main/java/org/apache/druid/sql/calcite/rule/FilterJoinExcludePushToChildRule.java

```diff
@@ -292,4 +298,42 @@ private static boolean classifyFilters(List<RexNode> filters,
     // Did anything change?
     return !filtersToRemove.isEmpty();
   }
+
+  private void removeRedundantIsNotNullFilters(List<RexNode> joinFilters, JoinRelType joinType)
+  {
+    if (joinType != JoinRelType.INNER) {
+      return; // only works for inner joins
+    }
+
+    ImmutableList.Builder<RexNode> isNotNullFiltersBuilder = ImmutableList.builder();
+    ImmutableList.Builder<Pair<RexNode, RexNode>> equalityFiltersOperandBuilder = ImmutableList.builder();
+
+    for (RexNode joinFilter : joinFilters) {
+      List<RexNode> operands = ((RexCall) joinFilter).getOperands();
+      if (joinFilter.isA(SqlKind.IS_NOT_NULL)) {
+        isNotNullFiltersBuilder.add(joinFilter);
+      }
+      if (joinFilter.isA(SqlKind.EQUALS) && operands.size() == 2) {
```

Comment:
should this be an else if?

Reply:
I think just `if` should also work since they are different kinds according to calcite. But changed the code to else if for better readability.